services/order.py

```diff
@@ -0,0 +1,43 @@
+from datetime import datetime
+from django.db import transaction
+
+from django.contrib.auth import get_user_model
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket, MovieSession
+
+
+@transaction.atomic
+def create_order(tickets: list[dict], username: str, date: str = None) -> None:
+    if not get_user_model().objects.filter(username=username).exists():
+        user = get_user_model().objects.create_user(username=username)
+    else:
+        user = get_user_model().objects.get(username=username)
```

Comment:
Consider handling the exception when a user with the given username does not exist. Although this is checked before, using a try-except block here can prevent potential race conditions in a concurrent environment.